connectors/agent/config.py

```diff
@@ -69,33 +76,92 @@ def try_update(self, source):
         the method returns False.
         """
 
+        source = unit.config.source
+
         # TODO: find a good link to what this object is.
         has_hosts = source.fields.get("hosts")
         has_api_key = source.fields.get("api_key")
         has_basic_auth = source.fields.get("username") and source.fields.get("password")
+
+        assumed_configuration = {}
+
+        # Log-related
+        assumed_configuration["service"] = {}
```

Comment:
I think it clears the `service._use_native_connector_api_keys` setting

Reply:
The root cause is how we were merging the two dicts with default python `update` https://github.com/elastic/connectors/blob/4f6ca9bb4d9d05f8c400252d717088a7734f4b99/connectors/agent/config.py#L116

This would override the whole `service` object from the default config stored in this the `agent/config.py` class (so this default flag `service._use_native_connector_api_keys` would be cleared)

I'm fixing this in this PR